include/zephyr/rtio/rtio.h

```diff
@@ -810,15 +804,26 @@ static inline void rtio_sqe_drop_all(struct rtio *r)
  */
 static inline struct rtio_cqe *rtio_cqe_consume(struct rtio *r)
 {
+	struct rtio_mpsc_node *node;
+	struct rtio_cqe *cqe = NULL;
+
 #ifdef CONFIG_RTIO_CONSUME_SEM
 	if (k_sem_take(r->consume_sem, K_NO_WAIT) == 0) {
-		return rtio_spsc_consume(r->cq);
-	} else {
-		return NULL;
+		node = rtio_mpsc_pop(&r->cq);
+		if (node == NULL) {
+			return NULL;
+		}
+		cqe = CONTAINER_OF(node, struct rtio_cqe, q);
```

Comment:
Missing `r->cqe_pool_used--;`? Overall the two branches of the `ifdef` seem very similar, maybe we can have a
```
static inline int rtio_lock(struct rtio *r)
{
#ifdef CONFIG_RTIO_CONSUME_SEM
    return k_sem_tak(r->consume_sem, K_NO_WAIT);
#else
    return 0;
#endif
}
```